src/hatch_rest_api/riot.py

```diff
@@ -178,9 +181,53 @@ def turn_clock_off(self):
     # favorite_name_id is expected to be a string of name-id since name alone isn't unique
     def set_favorite(self, favorite_name_id: str):
         _LOGGER.debug(f"Setting favorite: {favorite_name_id}")
-        fav_id = int(favorite_name_id.split("-")[1])
+        fav_id = int(favorite_name_id.rsplit("-", 1)[1])
         self._update({"current": {"srId": fav_id, "step": 1, "playing": "routine"}})
 
+    def set_audio_track(self, audio_track: RIoTAudioTrack):
+        _LOGGER.debug(f"Setting audio track: {audio_track}")
+        if audio_track == RIoTAudioTrack.NONE:
+            self.turn_off()
+            return
+        # Hard-coded list, as some of these values are not returned by the 'sounds' API. These were found from manually browsing the app and playing each
+        # song, collecting the necessary values (name, id, url) from the Home Assistant debug logs for the `ha_hatch` custom component integration.
```

Comment:
> Can this mapping be moved into const?

It could, yes. Would the intent/desire then be to copy the map here and update as needed? Or would you prefer that the 'const' itself then get updated as the below logic gets called?

> And should this be just keyed off url since the last pr had notes about the id being ignored, maybe url is better to key the enum on?

I'm personally not a fan of that, to be quite honest (though, I'm also not sure exactly what you have in mind - would the map be inverted? If so, that seems strange to me as what we want to lookup is the audio URL from the track name / id). My rational is that keying off the URL is much longer, and does not seem as clean.

> ...since the last pr had notes about the id being ignored...

What's odd about the 'id being ignored' part from the previous PR is that, in my testing, I've not noticed that (id is ignored) to be the case all of the time. Specifically, I tested this change w/ my setup (a) with and (b) without `id` being sent - the calls seemingly always worked with `id`, whereas sometimes (if I had to ballpark, it was maybe ~30% from what I recall now) the calls did not work without `id`. Because of this, I opted to leave it in there - I figured worst-case in keeping `id` as part of the call, Hatch API seems to be ignoring it and will still function as intended/expected. Worse-case in removing `id` was that the calls didn't seem to work (no error, from what I recall), which was rather frustrating.

Reply:
My thinking was something like https://stackoverflow.com/questions/71569320/python-enum-how-to-get-enum-values-with-multiple-attributes with the enum attributes being id and url to keep all the three values together vs having two two lookups which can lead to a sound being in one list but not the other and just more to track from a dev mental load.

As for the const, my goal is just to remove the large definitions from this method so its logic is easier to read.  You could put the hard coded list into a file that defined the multi attribute enum instead of const as it isn't really a const since it can be updated.

If you would rather keep the two separate lookups you can put the hard coded enum definition and initial map into const and make a copy of the map for modifying in this method.

As for the hard coded map helping in cases of hatch updating their urls returned from the API; that raises a concern I didn't think of, but if they updated the url then the wav files would likely no longer be there? If so the url being in the code seems more likely to cause instability not less.

Reply:
when I mentioned they might update the urls, my point was the wav files maybe taken down causing these hard coded urls to be inaccessible until this code is updated, it's a stability liability, but I don't have a strong opinion either way.

Reply:
Gotcha - yes, that is a valid and real concern. I'm not sure how to best handle that given what I've seen thus far. tbh, I'm not a huge fan of having the URLs in there to begin with (as I mentioned, it was working fine without the URLs being passed to the API before). Part of me wants to make another fork that just relies on passing the IDs to the API (and not the wav URLs) and see if I can observe it working fine again, or if they removed that ability entirely w/in the course of my development here.

Reply:
You could just add another method that sets it without the url, being able to play custom wav files through url seems useful, but if you can play without urls that's a good use case, you just have two methods.